src/CommunityToolkit.Maui.Core/Capabilities/StatusBar/StatusBar.android.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Diagnostics;
+using Android.OS;
+using Android.Util;
+using Android.Views;
+using Microsoft.Maui.Platform;
+using Activity = Android.App.Activity;
+using Debug = System.Diagnostics.Debug;
+
+namespace CommunityToolkit.Maui.Core.Capabilities;
+static partial class StatusBar
+{
+	static void PlatformSetColor(Color color)
+	{
+		if (!IsSupported())
+		{
+			return;
+		}
+		Activity.Window?.SetStatusBarColor(color.ToPlatform());
+	}
+
+	static void PlatformSetStyle(StatusBarStyle style)
+	{
+		if (!IsSupported())
+		{
+			return;
+		}
+
+		switch (style)
+		{
+			case StatusBarStyle.DarkContent:
+				AddBarAppearanceFlag(Activity, (StatusBarVisibility)SystemUiFlags.LightStatusBar);
+
+				break;
+			default:
+				RemoveBarAppearanceFlag(Activity, (StatusBarVisibility)SystemUiFlags.LightStatusBar);
+				break;
+		}
+	}
+
+
+	static void AddBarAppearanceFlag(Activity activity, StatusBarVisibility flag) =>
+		SetBarAppearance(activity, barAppearance => barAppearance |= flag);
+
+
+	static void RemoveBarAppearanceFlag(Activity activity, StatusBarVisibility flag) =>
+		SetBarAppearance(activity, barAppearance => barAppearance &= ~flag);
+
+
+	static void SetBarAppearance(Activity activity, Func<StatusBarVisibility, StatusBarVisibility> updateAppearance)
+	{
+		var window = GetCurrentWindow(activity);
+#pragma warning disable CS0618 // Type or member is obsolete
+		var appearance = window.DecorView.SystemUiVisibility;
+		appearance = updateAppearance(appearance);
+		window.DecorView.SystemUiVisibility = appearance;
+#pragma warning restore CS0618 // Type or member is obsolete
+
+		static Window GetCurrentWindow(Activity activity)
+		{
+			var window = activity.Window ?? throw new NullReferenceException();
+			window.ClearFlags(WindowManagerFlags.TranslucentStatus);
+			window.AddFlags(WindowManagerFlags.DrawsSystemBarBackgrounds);
+			return window;
+		}
+	}
+
+	static bool IsSupported()
+	{
+		if (PlatformVersion.SdkInt < (int)BuildVersionCodes.M)
```

Comment:
On Android doesn't make sense, because the APIs are different.

Reply:
Can we have a method/property isAndroidMOrNewer?

Reply:
I mean, why do we want to add more complexity? IMHO, since this code is just in Android world (`.android` file) doesn't add value trying to make it similar to iOS. They are two different worlds.

Reply:
If we use it only in 1 place do we need PlatformVersion class at all?